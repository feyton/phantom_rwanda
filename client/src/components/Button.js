import React from 'react';

const Button = ({ name, styles, onClick }) => {
  return (
    <div>
      <button
        type="submit"
        className={`${styles} rounded-xl py-2 text-base md:text-sm px-5`}
        onClick={onClick}
      >
        {name}
      </button>
    </div>
  );
};
export const ButtonA = ({ name, onClick, styles, type }) => {
  return (
    <div>
      <button
        type={type}
        onClick={onClick}
        className={`${styles} bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 `}
      >
        {name}
      </button>
    </div>
  );
};

export default Button;

export const ButtonLoading = ({ name, styles }) => {
  return (
    <div className="flex w-fit py-1 px-3 bg-primary font-raleway font-bold rounded-md">
      <div className="loader mr-1" />
      <button className="ml-1 font-raleway text-white font-bold">
        {' '}
        {name}
      </button>
    </div>
  );
};

export const PrimaryButton = ({ name, type = 'button', ...options }) => {
  return (
    <button
      {...options}
      type={type}
      className="bg-primary rounded-md text-white font-bold font-raleway py-2 px-3 hover:bg-hover transition-all block hover:transition-all"
    >
      {name}
    </button>
  );
};
