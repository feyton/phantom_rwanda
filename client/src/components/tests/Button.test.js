import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from '../Button.js';

const button = {
  name: 'start',
  email: 'bg-blue-300'
};

describe('Button', () => {
  it('it should render Button', () => {
    const btn = renderer.create(<Button />).toJSON();
    expect(btn).toMatchSnapshot();
  });
  it('accepts button props', () => {
    const wrapper = mount(<Button props={button} />);
    expect(wrapper.props().props).toEqual(button);
  });
  it('contains the button element', () => {
    const wrapper = mount(<Button props={button} />);
    const value = wrapper.children().length;
    expect(value).toEqual(1);
  });
});
