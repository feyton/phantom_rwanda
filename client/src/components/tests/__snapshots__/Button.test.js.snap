// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it should render Button 1`] = `
<div>
  <button
    className="undefined rounded-xl py-2 text-base md:text-sm px-5"
    type="submit"
  />
</div>
`;
